op-supervisor/supervisor/backend/db/db.go

```diff
@@ -51,9 +51,17 @@ func NewChainsDB(logDBs map[types.ChainID]LogStorage, heads HeadsStorage) *Chain
 	}
 }
 
+func (db *ChainsDB) AddLogDB(chain types.ChainID, logDB LogStorage) {
+	if db.logDBs[chain] != nil {
+		log.Warn("overwriting existing logDB for chain", "chain", chain)
+	}
+	db.logDBs[chain] = logDB
+}
+
 // Resume prepares the chains db to resume recording events after a restart.
 // It rewinds the database to the last block that is guaranteed to have been fully recorded to the database
 // to ensure it can resume recording from the first log of the next block.
+// TODO: I am interested in renaming this to something more descriptive like "PrepareWithRollback"
```

Comment:
```suggestion
// TODO(#11793): we can rename this to something more descriptive like "PrepareWithRollback"
```
https://github.com/ethereum-optimism/optimism/issues/11793